crates/libgraphql-core/src/test/golden_tests/test_runner.rs

```diff
@@ -0,0 +1,632 @@
+use crate::ast;
+use crate::operation::ExecutableDocumentBuilder;
+use crate::operation::FragmentRegistryBuilder;
+use crate::schema::Schema;
+use crate::schema::SchemaBuilder;
+use crate::test::golden_tests::golden_test_case::GoldenTestCase;
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+
+#[cfg(test)]
+use rayon::prelude::*;
+
+/// Result of a single golden test
+#[derive(Debug)]
+pub struct GoldenTestResult {
+    pub test_name: String,
+    pub passed: bool,
+    pub error_message: Option<String>,
+    pub file_path: PathBuf,
+    pub file_snippet: Option<String>,
+}
+
+/// Collection of golden test results
+#[derive(Debug)]
+pub struct GoldenTestResults {
+    pub results: Vec<GoldenTestResult>,
+}
+
+impl GoldenTestResults {
+    pub fn new() -> Self {
+        Self {
+            results: Vec::new(),
+        }
+    }
+
+    pub fn all_passed(&self) -> bool {
+        self.results.iter().all(|r| r.passed)
+    }
+
+    pub fn extend(&mut self, results: Vec<GoldenTestResult>) {
+        self.results.extend(results);
+    }
+
+    pub fn failure_report(&self) -> String {
+        let failures: Vec<_> = self.results.iter().filter(|r| !r.passed).collect();
+
+        format!(
+            "{} of {} golden tests failed:\n\n{}",
+            failures.len(),
+            self.results.len(),
+            failures
+                .iter()
+                .map(|r| format_detailed_failure(r))
+                .collect::<Vec<_>>()
+                .join("\n\n")
+        )
+    }
+
+    pub fn summary(&self) -> String {
+        let all_passed = self.all_passed();
+        let emoji = if all_passed { "✅" } else { "❌" };
+        let banner = format!("{emoji} ========================================");
+        let header = format!("{emoji} GOLDEN TEST SUMMARY");
+
+        if all_passed {
+            format!(
+                "{}\n{}\n{}\nTotal tests: {}\nPassed: {}\nFailed: 0\n\nAll golden tests passed!\n{}",
+                banner, header, banner, self.results.len(), self.results.len(), banner
```

Comment:
Inline your variables in the format string here like:
```suggestion
                "{banner}\n{header}\n{banner}\nTotal tests: {}\nPassed: {}\nFailed: 0\n\nAll golden tests passed!\n{banner}",
               self.results.len(), self.results.len()
```

Reply:
[claude]: Agreed\! I'll inline the variables in the format string as suggested. This follows Rust best practices and the clippy lint recommendations.